Sources/Articles/Documentation.docc/2024/how-to-enable-http2-in-hummingbird/how-to-enable-http2-in-hummingbird.md

```diff
@@ -0,0 +1,51 @@
+# Enabling HTTP/2 in Hummingbird
+
+[Hummingbird](https://hummingbird.codes) features a flexible architecture that allows swapping various components in and out. This flexibility extends to the networking layer, where you can easily swap out the default HTTP/1 networking code for one supporting HTTP/2.
+
+## Package Manifest
+
+As part of your code, you'll have a target that depends on Hummingbird.
+
+```swift
+.executableTarget(
+    name: "App",
+    dependencies: [
+        .product(name: "Hummingbird", package: "hummingbird"),
+        ...
+    ]
+),
+```
+
+Because Hummingbird aims to only compile what you need, Hummingbird doesn't enable HTTP/2 support by default. In order to use HTTP/2, you'll first need to depend on the `HummingbirdHTTP2` module within the Hummingbird package.
+
+Add the following product to your target's dependencies:
+
+```swift
+.product(name: "HummingbirdHTTP2", package: "hummingbird"),
+```
+
+## Enabling HTTP/2
+
+First, in your application, you'll need to import the relevant dependencies:
+
+@Snippet(path: "site/Snippets/hummingbird-2-http-2", slice: "imports")
+
+HTTP/2 can only be enabled if your server already supports TLS. In order to support TLS, you'll need to create a ``TLSConfiguration`` from ``NIOSSL``.
+
+You'll need to obtain this certificate through a certificate authority (CA) or generate a self-signed certificate.
```

Comment:
Could you add a short section on how to generate a self-signed certificate and key? That would be really useful, in my opinion.

FYI: https://theswiftdev.com/a-simple-http2-server-using-vapor-4/#certificate-generation

Reply:
@tib would you want to write that section? I'd prefer not using RSA in my the recommended certificates of course.

Reply:
I can do that, what's your recommended algo? 🤔

Reply:
Any elliptic curve